play-ws-standalone/src/main/java/play/libs/ws/BodyWritable.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package play.libs.ws;
+
+import java.util.function.Supplier;
+
+/**
+ * Writes out a {@code WSBody<A>}
+ *
+ * @param <A> the type of body.
+ */
+public interface BodyWritable<A> extends Supplier<WSBody<A>> {
```

Comment:
Not saying that needs to be implemented now but would be interested in getting your thoughts on if that'd be eventually possible to write a single implementation that works for Play action and WS bodies.